### PR TITLE
Lossentropy

### PR DESCRIPTION
**Pull request by isaactony**
## Summary by CodeRabbit

- **New Features**
  - Added a CSV cleaning tool that automatically removes incomplete records.
  - Introduced a basic authentication mechanism using preset credentials.
  - Enhanced data processing with consistent text encoding.
  - Improved predictive performance with a custom loss function.
  - Upgraded external data retrieval with robust error handling and reliable fallback responses.

### CODE REVIEW

File: lossentropy.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import tensorflow as tf
+
+def custom_loss(y_true, y_pred):
+    """    
+    Positive cases (y_true = 1) are given higher weight to improve sensitivity.
+    """
+    weights = tf.where(y_true == 1, 25.0, 1.0)  # Give rare cases more importance
+    return tf.reduce_mean(weights * (y_true - y_pred)**2)  # Weighted MSE loss
```
[the remark this answers is not in the document]
Comment (isaactony) [reply lossentropy.py:8]: @coderabbitai full review and sugges use of built-in loss function like BinaryCrossentropy(), arguing that sticking to standard implementations improves reliability and maintainability.
